### ISSUE TITLE
Utilize 'view' for Synapse projects

### ISSUE THREAD

**Issue by cconrad8** [nf-osi/jobs#74]
![image](https://github.com/nf-osi/jobs/assets/114612268/aebc742a-f2a3-47fc-96bd-9dd27fb0a6c4)

Currently, we utilize the portal-studies table to change the data status on the nf data portal https://www.synapse.org/#!Synapse:syn16787123/tables/ 

however, this does not sync with the annotations at the project level 

![image](https://github.com/nf-osi/jobs/assets/114612268/6c616a64-1383-4e80-beb5-ff2777714f9d)

here, data status says "under embargo" but actually it is available.  

1. Add studies to the scope of a newly created project view. (similar to how we do files I think) 
2. Make sure new synid table is updated.  

Outcome - information will be matched and could help with changing data status more easily down the line

**Comment by cconrad8** [comment]
tagging @anngvu @allaway @jaybee84

**Comment by allaway** [comment]
There's one technical limitation here: the max length of an annotation is 500 characters, but the project summaries are often much longer. This is actually why we went with a Table instead of a Project View (we were going to do that initially for the reasons you mentioned - easier to keep everything synced up). I would consider the current project-level annotations to generally be out of date/not the gold standard, unless maybe someone has synced them up recently.

**Comment by allaway** [comment]
I should rephrase: The max length of an annotation in 2019 was 500 characters. I am pretty sure that is unchanged. There may be a workaround, such as the annotation referencing an external wiki page that contains the full text, but we would have to see if the SRC can do that.

**Comment by anngvu** [comment]
@cconrad8 So this idea did come up. In addition to what Robert mentioned regarding the character limit, here are other reasons. So reason 1 and 2 still apply, 3 has been resolved recently with some reorg of the DHART-SPORE projects. Reason 1 maybe doesn't matter so much, though, but we still have to figure out reason 2 before I start on this.

<img width="1131" alt="image" src="https://github.com/nf-osi/jobs/assets/32753274/f78cae6c-a56c-45c7-bebf-afba4aad3fb4">

**Comment by anngvu** [comment]
Hmm, actually, glad this was brought up again, because something else might enable this (materialized views and possibly virtual tables). What if we do use annotations for everything except that pesky summary? We maintain summary in a regular table, and full studies could just be a materialized view/virtual table combining both.

**Comment by allaway** [comment]
🤯 I forgot we could do joins. That seems like a great solution! We should just check with Jay to make sure that this will work seamlessly with the SRC.

**Comment by jaybee84** [comment]
@anngvu using materialized views if possible is a great idea. 

It still requires us to update the summary table separately during project intake, but hopefully its a one time thing with the summaries being more static than other annotations :)  

@allaway what is SRC in this context?

**Comment by allaway** [comment]
> @allaway what is SRC in this context?

Sorry, Synapse React Client :)

**Comment by cconrad8** [comment]
Good idea @anngvu And we could probably just leave the portal studies page - remove all the other stuff besides synapseid and project summary https://www.synapse.org/#!Synapse:syn16787123/tables/ so one table is already almost made!

**Comment by cconrad8** [comment]
@jaybee84 

> It still requires us to update the summary table separately during project intake, but hopefully its a one time thing with the summaries being more static than other annotations :)

I'm wondering if you can explain here what you mean by update summary table separately? 
I was thinking the summaries would still be added automatically to a Synapse table

**Comment by allaway** [comment]
>Good idea @anngvu And we could probably just leave the portal studies page - remove all the other stuff besides synapseid and project summary https://www.synapse.org/#!Synapse:syn16787123/tables/ so one table is already almost made!

Yes, this would be ideal as opposed to creating a new table because this synapse ID is already whitelisted to be anonymously accessible (i.e. even people who are not logged in can see the contents) - it's a whole process right now to make data anonymously accessible.

**Comment by jaybee84** [comment]
> I'm wondering if you can explain here what you mean by update summary table separately?
> I was thinking the summaries would still be added automatically to a Synapse table

@cconrad8 maybe I am missing something. How are the summaries getting added to the table currently? Maybe there is a DCC workflow that I missed here

**Comment by cconrad8** [comment]
@jaybee84 Ah I believe when the project is provisioned it updates the table syn16787123
in this script: 
https://github.com/nf-osi/jobs/blob/main/new_project/new_project_from_json.R

**Comment by jaybee84** [comment]
Perfect! That makes this solution even more lucrative!
